### DISCUSSION TITLE
need help with either someone compiling marlin for me or teaching me so i can change my bed size to 400 X 400 X 450

### DISCUSSION THREAD

**Discussion by blackoutlaw** [Jyers/Marlin#905] [Q&A] [answered]
Hello all I have just acquired the Ender Extender 400XL Kit for my E3V2 and am in need of changing my bed size in marlin and CURA so I can utilize the new print area.  Every time I compile the code I get errors with it and then Visual studio crashes.  I can compile the source code when I first get it just fine without any changes but when I change just the numbers for the bed size then save and compile I get all errors with the lcd lines and it crashes.  Any Help would be greatly appreciated with this issue.  Either with someone able to compile the lasted Jayers firmware for the 4.2.2 with BL Touch or giving me some pointers on how i can do it myself.  Thanks

**Comment by Cal-LF** [comment, marked as answer]
Thought I would try to compile the BL Touch 3X3 template. It compiled successfully.
If you would like to try it, here is the .bin on my MS Onedrive. https://1drv.ms/u/s!Ahj7yiz5K62hhg14FZEmzqvQwEop?e=80UEts

![Marlin](https://user-images.githubusercontent.com/76499230/119582122-06175780-bd92-11eb-8fb7-4aee72584804.jpg)
![Bed Size](https://user-images.githubusercontent.com/76499230/119582559-bd13d300-bd92-11eb-98a4-cbbe8d2dc1d3.jpg)

**Reply by blackoutlaw** [reply]
thank you does this work as the current extendUI that jayers May 1st update is

**Reply by blackoutlaw** [reply]
also would it be possible to do it in a bl touch 15 x 15 on the new bed size or is 10 x 10 the max. only asking because of the size of the bed i think the mesh will need more points to be more accurate

**Reply by Cal-LF** [reply]
This is the current release. I will put the 10 x 10 UBL and try 15 x 15 and put that up too.

**Reply by blackoutlaw** [reply]
ubl is that the bltouch or just ubl. what is ubl

**Reply by Cal-LF** [reply]
Unified bed leveling. Gives larger grid options. In this case the firmware notes that 15x15 is the max.
I don't know if these will work but they did compile successfully. I don't use BL touch so have never tried compiling these before.

![UBL](https://user-images.githubusercontent.com/76499230/119587585-f94c3100-bd9c-11eb-8d23-ac7ac55d0f00.jpg)

**Reply by blackoutlaw** [reply]
ok so does the ubl use the bltouch like the mesh leveling and if so then perfect with the 15 x 15 and can that be hs like jayers 10x10 hs im kinda new to firmware im an old school cobalt programmer

**Reply by Cal-LF** [reply]
I am unsure where to activate the high speed option.. These will likely not be high speed probing.
I use manual mesh 4x4 and it works well for me.

**Reply by blackoutlaw** [reply]
that is ok.  how do you do the 4 x 4 currently i can only do the 3 x 3 with my standard bed and the 3 x 3 firmware did you compile the 4 x 4 and that is why you can do a 4 x 4 manual

**Reply by Cal-LF** [reply]
Yes, I compile from the 3X3 manual mesh template. Configuration.h Line 1452 #define GRID_MAX_POINTS_X 3 4 or 5

These are the changes I make with each update. 
[Jyres Compile Changes.txt](https://github.com/Jyers/Marlin/files/6543247/Jyres.Compile.Changes.txt)

**Reply by blackoutlaw** [reply]
ok what is the max manual mesh that can be set 5 x 5 or 10x10

**Reply by Cal-LF** [reply]
I asked that question. #897 
Have not tried over 5x5. Thinking 10x10 but that is a lot of paper sliding.
UBL is stored on eprom and saves with updates, manual mesh is not. It erases when updating firmware.

**Reply by blackoutlaw** [reply]
so that is great to know about the ubl

**Reply by Cal-LF** [reply]
Update after trying the firmwares. I am curious to know if they work.

**Reply by blackoutlaw** [reply]
i will update it will be about a week before i have the kit installed and set the way i want  so will you be posting a 15x15 and a 10x10 with the 5x5 manual mesh

**Reply by blackoutlaw** [reply]
thank you for all your support were you able to do the 10 x 10 or the 15 x 15

**Reply by Cal-LF** [reply]
Yes, they are on my Onedrive link above. I'll ad the 5x5 manual there also.

**Reply by blackoutlaw** [reply]
everytime i go to that link i only see the 3 x 3 you did first and gave the link

**Reply by Cal-LF** [reply]
Sorry, this link is to the folder.
https://1drv.ms/u/s!Ahj7yiz5K62hhUG4dcMUmHOeiOGD?e=oFxTFt

**Reply by blackoutlaw** [reply]
thanks one last question is this the one for the 4.2.2 board

**Reply by blackoutlaw** [reply]
only asking because my printer has the 4.2.2 board for some odd reason but has the quite steppers

**Reply by Cal-LF** [reply]
Yes. I listed that in the details of each file. Right click on the Onedrive will let you see that.
V422 is a quiet board as is the V427.

**Reply by Cal-LF** [reply]
This is an instruction to compile with VS Code.
[Compiling Steps.txt](https://github.com/Jyers/Marlin/files/6549478/Compiling.Steps.txt)

**Reply by blackoutlaw** [reply]
thank you for all your support I will post an update to this discussion when I try the firmware out

**Reply by smg492** [reply]
> Sorry, this link is to the folder.
> https://1drv.ms/u/s!Ahj7yiz5K62hhUG4dcMUmHOeiOGD?e=oFxTFt

I gave this a shot, and the extruder put out 110mm of filament when testing esteps.  I cannot change these in a bin file correct?

**Reply by Cal-LF** [reply]
Esteps are adjusted in the Control menu. Motion - Step/mm - Extruder
I have mine set at 97. 93 is stock Marlin setting.

**Comment by smooreace** [comment]
Yes, it can be pre-set in configuration-h

**Reply by Cal-LF** [reply]
Yes, I pre-set several settings in the config files as in the text file I sent earlier. If you change any settings in the menus, save to eprom.

**Comment by blackoutlaw** [comment]
ok i finally got the upgrade to bed size done and flashed the above 10 x 10  ubl firmware all seams to be doing good except after i do the generate a mesh it only does about 60 points and then says that it got all accessible  points and it does a random pattern when doing the mesh and when viewing the mesh it is not consistent then i do the auto tilt and it gets worse as in it causes some points not to be within the .1 of the bed and others to drive the nozzle into the bed.  is there an issue with the firmware or is it a physical thing i have done all the leveling via the lcd screen menues . could someone let me know if i am doing anything wrong.  at this point i have not changed my start gcode in cura so if there is something i need to do that would be a great help. as my bed is now a 400 x 400x 450 from the original 3v2 bed size.  thanks for any help

**Comment by smooreace** [comment]
There are hundreds of videos out there on compiling your own firmware, and
several on bed size specifically. Its nothing more than editing a text file

**Comment by blackoutlaw** [comment]
i have the firmware and it is loaded but the bed leveling is not acting right i just want to know if this is something im doing wrong with the printer or if im missing a step i should be doing prior to leveling or after i level as the bed leveling is giving me points that are to close and some that are to far away and it does not even do a complete 10 x 10 level

**Comment by fabmce** [comment]
idem pour moi 
firmware UBL bl touch 10x10 et il me construit un mesh de 8x8 (64 points) et il m'affiche tout de même une view de 100 pts ?
